src/WebJobs.Script.WebHost/Middleware/SystemTraceMiddleware.cs

```diff
@@ -18,16 +18,19 @@ internal class SystemTraceMiddleware
     {
         private readonly ILogger _logger;
         private readonly RequestDelegate _next;
+        private readonly HostNameProvider _hostNameProvider;
 
-        public SystemTraceMiddleware(RequestDelegate next, ILogger<SystemTraceMiddleware> logger)
+        public SystemTraceMiddleware(RequestDelegate next, ILogger<SystemTraceMiddleware> logger, HostNameProvider hostNameProvider)
         {
             _logger = logger;
             _next = next;
+            _hostNameProvider = hostNameProvider;
         }
 
         public async Task Invoke(HttpContext context)
         {
             SetRequestId(context.Request);
+            _hostNameProvider.Synchronize(context.Request);
```

Comment:
Is this done on every request to catch changes from swaps?

Reply:
Correct, as detailed in the linked issue https://github.com/Azure/azure-functions-host/issues/4162